### ISSUE TITLE
Post request expects Unit for Body

### ISSUE THREAD

**Issue by santoshastagi** [ktorio/ktor#904]
### Feedback

While trying a post request as follows:
```
httpClient.call {
      method = HttpMethod.Post
      url("/postUrl")
      contentType(ContentType.Application.Json)
}
```
I omitted the body, because I did not have any body. I got the following error message:

```
kotlinx.serialization.SerializationException: Can't locate argument-less serializer for class io.ktor.client.utils.EmptyContent. For generic classes, such as lists, please provide serializer explicitly.
```

After trying a few things, I tried Unit for body and the request succeeded. But the error message was not helpful to figure that out. I'd expect an error message on the lines of "Missing body"

OR atleast in the case when the post request does not specify a body could it internally default to Unit?

**Comment by e5l** [comment]
Hi @santoshastagi,  thanks for the report. 

The fix is here and would be available in `1.2.0`.

**Comment by ScottPierce** [comment]
@e5l When will 1.2.0 be available? My team is interested in a few fixes that are in for 1.2.0
